test/chrome/crdpMultiplexing/crdpMultiplexor.test.ts

```diff
@@ -0,0 +1,175 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import { CRDPMultiplexor } from '../../../src/chrome/crdpMultiplexing/crdpMultiplexor';
+import { LikeSocket } from 'noice-json-rpc';
+import { Mock, Times, It } from 'typemoq';
+import * as Assert from 'assert';
+
+class StubSocket implements LikeSocket {
+
+    send(message: string): void { /* empty */ }
+    removeListener(event: string, cb: Function): any { /* empty */ }
+    on(event: 'open', cb: (ws: LikeSocket) => void): any;
+    on(event: 'message', cb: (data: string) => void): any;
+    on(event: string, cb: Function): any { /* empty */ }
+}
+
+suite('CRDPMultiplexor', () => {
+
+    let multiplexor: CRDPMultiplexor;
+    let webSocketMock: Mock<StubSocket>;
+    let socketMessageCallbacks: Function[];
+
+
+    setup(() => {
+        let socket = new StubSocket();
+        socketMessageCallbacks = [];
+        webSocketMock = Mock.ofInstance<StubSocket>(socket);
+        webSocketMock
+            .setup(s => s.on('message', It.isAny()))
+            .returns((s: string, cb: any) => {
+                socketMessageCallbacks.push(cb);
```

Comment:
Why not implement this directly inside of StubSocket  instead of using a Mock? You can store all the received or sent messages in an array if you want to verify them